### PR TITLE
Annotation fixes

### PR DESCRIPTION
**Pull request by AOS002**
- If no simplifications are done when eliminatable=Fale & relocatable=True, then the annotations still need to pass from the args to the new AST
- The annotations were not being passed through to the results for the claripy.If()

### CODE REVIEW

File: claripy/ast/base.py [hunk 1]
```diff
@@ -272,6 +272,11 @@ def __a_init__(self, op, args, variables=None, symbolic=None, length=None, simpl
                 tuple(a for a in self.annotations if not a.eliminatable and a.relocatable)
             ))).keys()
 
+            self.annotations = tuple(itertools.chain(
```
Comment (zardus) [inline claripy/ast/base.py:275]: I might be missing something, but doesn't the above already catch eliminatable=False, relocatable=True annotations?

Comment (AOS002) [reply claripy/ast/base.py:275]: Yes that is correct, but they do not pass on the annotations from the `ast_args` to the resultant AST they only set the `self._uneliminatable` & `self._relocatable`. I am assuming that if `relocatable=True` and `eliminatable=False/True`, then the annotations from the arguments should be relocated to the annotations of the new AST(i.e. `self.annotations`).

**Comment by zardus** [comment]
I forgot about this until now! Fish convinced me that it is correct. Are we good to merge?

**Comment by AOS002** [comment]
I think we are good to merge! although I see that there is one test that's failing. The same test is failing on the most recent commit as well, so I don't think it should be because of the changes I made.

**Comment by rhelmot** [comment]
I'm hitting merge on this. The CI run has vanished into the ether, but based on what was said I'm gonna assume it was test_unsat_core failing, which is known nondeterministic.

**Pull request merged**